reference input feature (#4074)
input-output-hk/cardano-wallet: specifications/api/mint-burn.md

The "Transactions New > Construct" HTTP endpoint allows the creation and use of reference inputs.

Specifically:

1. Creation of a transaction output that contains a minting script and is suitable for use as reference input.

    In the `reference_policy_script_template` field, you can optionally add a script template. The HTTP endpoint will map this script template into a script using the wallet's policy public key, and this script will be included in the first transaction output (i.e. at index `0`) of the transaction.

    Example `POST` data for the endpoint:

    ```
    {
    ...
      "reference_policy_script_template":
          { "all":
             [ "cosigner#0",
               { "active_from": 120 }
             ]
          },
    ...
    }
    ```

2. Using a refence input that contains a minting script.

    In the `mint_burn` field, the array element contains `reference_input`. This field specifies a transaction input (pair of transaction ID and output index) which will be added as a reference input and is assumed to contain the minting script. (If the corresponding output was created using the method above, the appropriate output index is `0`).

>     Example `POST` data for the endpoint:

Reviewer: ```suggestion
    Example `POST` data for the endpoint with reference input:
```

Author: done

    ```
    {
    ...
    "mint_burn": [{
        "reference_input":
            { "id": "464917d2bac71df96269c2d7c34dcb83183b8a3a3253c06e9d6a8bd0681422c9",
              "index": 0
            },
        "asset_name": "ab12",
        "operation":
            { "mint" :
                  { "receiving_address": #{destination},
                     "quantity": 10000
                  }
            }
    }]
    ...
    }
    ```

>     in constrast to example `POST` data for the endpoint when script is defined

Reviewer: ```suggestion
    For contrast: Example `POST` data for the endpoint with script template:
```

Author: done
